src/java/org/apache/cassandra/metrics/TableMetrics.java

```diff
@@ -1027,6 +953,35 @@ public Long getValue()
         return cfGauge;
     }
 
+    /**
+     * Same as {@link #createTableGauge(String, Gauge, Gauge)} but accepts a deprecated
+     * name for a table {@code Gauge}. Prefer that method when deprecation is not necessary.
+     *
+     * @param name the name of the metric registered with the "Table" type
+     * @param deprecated the deprecated name for the metric registered with the "Table" type
+     */
+    protected <G,T> Gauge<T> createTableGaugeWithDeprecation(String name, String deprecated, Gauge<T> gauge, Gauge<G> globalGauge)
```

Comment:
@dcapwell The only real difference is that it registers both new and deprecated metric names w/ the "Table"-based name factory. As an aside, I'm not sure why we would need to write the deprecated name w/ the `aliasFactory`. I think we should just remove that.

Reply:
Either way, this should all get removed in trunk once we cut `cassandra-4.0`.

Reply:
My hope is https://issues.apache.org/jira/browse/CASSANDRA-16082 would tell you if we need aliasFactory for deprecated names or not 😉  

though, Table was present in 3.0 so I think it would be needed here as well since ColumnFamlily and Table both existed in 3.0

Reply:
I'll just leave it for now.